pkg/server/api/agent/v1/service.go

```diff
@@ -398,6 +417,18 @@ func (s *Service) RenewAgent(ctx context.Context, req *agentv1.RenewAgentRequest
 // CreateJoinToken returns a new JoinToken for an agent.
 func (s *Service) CreateJoinToken(ctx context.Context, req *agentv1.CreateJoinTokenRequest) (*types.JoinToken, error) {
 	log := rpccontext.Logger(ctx)
+	parseRequest := func() logrus.Fields {
+		fields := logrus.Fields{}
+
+		if req.Ttl > 0 {
+			fields[telemetry.TTL] = req.Ttl
+		}
+		if req.Token != "" {
+			fields[telemetry.Token] = api.HashByte([]byte(req.Token))
```

Comment:
This isn't safe to log. If someone with permissions to access the server log intercepted this token before it was used, they could impersonate an agent.

Reply:
done, I removed token + payload uses